lib/src/main/java/com/auth0/android/lock/events/DatabaseSignUpEvent.java

```diff
@@ -43,33 +43,42 @@
 
     @NonNull
     private String password;
-    private Map<String, String> extraFields;
+    private Map<String, Object> rootAttributes;
 
     public DatabaseSignUpEvent(@NonNull String email, @NonNull String password, @Nullable String username) {
         super(email, username);
         this.password = password;
+        this.rootAttributes = new HashMap<>();
     }
 
     @NonNull
     public String getPassword() {
         return password;
     }
 
+    public void setRootAttributes(@NonNull Map<String, Object> attributes) {
+        this.rootAttributes.putAll(attributes);
+    }
+
+    /**
+     * Set the fields to set as user_metadata
+     *
+     * @param customFields user_metadata fields to set
+     */
     public void setExtraFields(@NonNull Map<String, String> customFields) {
-        this.extraFields = customFields;
+        this.rootAttributes.put(KEY_USER_METADATA, customFields);
```

Comment:
You store your metadata inside the rootAttributes dictionary?

Reply:
What I call here `rootAttributes` is a map of additional sign up parameters. And yes, metadata goes into them just like it was being sent before. The only difference is that before, metadata was the only attribute we would send. Now it has siblings :)